Coverage and Quality Report: Add reporting on use of forced-colors media query and currentcolor value

jongund opened the pull request:
The pull request updates the coverage script to:
1. Remove information about the use of `forced-color-adjust` from the report, since I thin we should not even be using it anymore in examples.
2. Adds information on examples using `forced-colors` media query.
3. Adds information on examples using `currentcolor` value for CSS properties that support color values.

[Preview of Updated Coverage Report](https://deploy-preview-324--aria-practices.netlify.app/aria/apg/about/coverage-and-quality/)

___
[WAI Preview Link](https://deploy-preview-324--aria-practices.netlify.app/ARIA/apg) _(Last built on Tue, 23 Jul 2024 06:01:44 GMT)._

mcking65 commented:
@jongund is this a replacement for #2902

jongund commented:
@mcking65 
Yes this would be the report we would want now.
Writing the high contrast support practices clarified for me the relevant information we needed in the report.

howard-e commented on scripts/coverage-report.js:
```diff
@@ -521,7 +521,9 @@ glob
       beforeCSS: getNumberOfReferences(dataCSS, '::before'),
       afterCSS: getNumberOfReferences(dataCSS, '::after'),
-      forcedColorAdjust: getNumberOfReferences(dataCSS, 'forced-color-adjust'),
+      //      forcedColorAdjust: getNumberOfReferences(dataCSS, 'forced-color-adjust'),
```
Could this be removed instead of commenting out?

howard-e commented on scripts/coverage-report.js:
```diff
@@ -1008,7 +1010,9 @@ let IndexOfExampleGraphics = indexOfExamples.reduce(function (set, example) {
   count += example.svgCSS;
   count += example.svgJS;
-  count += example.forcedColorAdjust;
+  //  count += example.forcedColorAdjust;
```
Same question as above since it should no longer be used.

jongund replied:
Yes, I can take it out

howard-e commented on scripts/coverage-report.js:
```diff
@@ -1125,7 +1140,9 @@ $('#example_mouse_pointer_tbody').html(IndexOfExampleMousePointer);
-$('#example_summary_force_color').html(countForcedColorAdjust);
+$('#example_summary_force_colors').html(countForcedColors);
+$('#example_summary_current_color').html(countCurrentColor);
+// $('#example_summary_force_color_adjust').html(countForcedColorAdjust);
```
Same question as above

howard-e commented on scripts/coverage-report.template:
```diff
@@ -168,9 +168,19 @@
+            <th scope="row">Uses <code>currentColor</code> value</th>
+            <td id="example_summary_current_color"></td>
+          </tr>
+<!--
+          <tr>
+            <th scope="row">Uses <code>forced-color-adjust</code> in CSS/SVG</th>
+            <td id="example_summary_force_color_adjust"></td>
+          </tr>
+-->
```
Same question as above

jongund replied:
I will take it out

howard-e reviewed (commented):
@jongund other than the questions in line, could you also merge this with `main` so https://github.com/w3c/aria-practices/pull/3026 changes could be included?

`npm run coverage-report` can be ran again afterwards to make sure things are in sync.

jongund commented:
@mcking65 
I also removed the CSS :before, :after and content columns in the table.  
Changed the name of the table to "SVG and High Contrast Techniques".  
The use of CSS :before and :after are not part of the coding practices.

mcking65 commented:
@howard-e 
Why is Check examples/index.html / coverage failing?

I assume that is why the preview still shows the CSS columns Jon removed from the graphics techniques table; the generation script didn't run?

I requested re-review now that Jon has made requested updates.

howard-e commented:
> @howard-e Why is Check examples/index.html / coverage failing?

This PR needs to manually re-run `npm run coverage-report` because the lint-staged step to automatically do it doesn't trigger when the generating script alone is updated in a PR. Instead, it was only being triggered when there are changes to `content/patterns/**/examples/*.html`.

But that is odd that the generating scripts weren’t being captured so I submitted #3048 to address it.

> I assume that is why the preview still shows the CSS columns Jon removed from the graphics techniques table; the generation script didn't run?

Not exactly, this was because there was an invalid file read happening with the new experimental content support on the build repo because of an uncaught assumption that the newly created files in #2977 were already present when the build repo references w3c/aria-practices.

Already submitted https://github.com/w3c/wai-aria-practices/pull/330 to fix and has been merged and and Jon’s PR preview has been regenerated

> I requested re-review now that Jon has made requested updates.

Will check it!

howard-e reviewed (commented):
@jongund thanks for addressing the feedback and the additional changes look good to me.

This still just needs a merge with `main` to include #3026, running `npm run coverage-report` and a commit and push to be ready.
